server/service.go

```diff
@@ -129,6 +146,62 @@ func (m *BoostService) handleStatus(w http.ResponseWriter, req *http.Request) {
 	fmt.Fprintf(w, `{}`)
 }
 
+// sendValidatorPreferences is used to send the validators preferences to the registered relays
+func (m *BoostService) sendValidatorPreferences(log *logrus.Entry, payload []types.SignedValidatorRegistration) uint64 {
+	// We need a wait group to manage each routine used to perform the requests.
+	var wg sync.WaitGroup
+
+	// Use an atomic counter to count successful requests.
+	numSuccessRequestsToRelay := uint64(0)
+
+	// Send the validators preferences to each registered relay.
+	for _, relay := range m.relays {
+		wg.Add(1)
+
+		go func(relayAddr string) {
+			defer wg.Done()
+
+			url := relayAddr + pathRegisterValidator
+			log := log.WithField("url", url)
+
+			err := SendHTTPRequest(context.Background(), m.httpClient, http.MethodPost, url, payload, nil)
+			if err != nil {
+				log.WithError(err).Warn("error in registerValidator to relay")
+				return
+			}
+
+			atomic.AddUint64(&numSuccessRequestsToRelay, 1)
+		}(relay.Address)
+	}
+
+	wg.Wait()
+
+	return numSuccessRequestsToRelay
+}
+
+func (m *BoostService) registerValidatorAtInterval(interval time.Duration, done chan bool) {
+	var payload []types.SignedValidatorRegistration
+	log := m.log.WithField("method", "registerValidatorAtInterval")
+
+	ticker := time.NewTicker(interval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-done:
+			// mev-boost has probably stopped
+			return
+		case payload = <-m.newRegistrationsRequests:
```

Comment:
this will overwrite the previous payload, but it's conceivable that there's multiple validators using one BN that would then overwrite each others registrations. perhaps better to make this additive?

Reply:
I'm actually not sure about it, the expectation is to have a single BN connecting to mev-boost, and forcing this to be additive would mean you cannot ever remove a registration.

Reply:
@Ruteri expectation is a single BN, but this single BN could have _many_ validator clients connected! We can not ignore this case.

One approach could be to add a time to each registration and remove old ones.